logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageSender.kt

```diff
@@ -145,15 +146,23 @@ internal class MessageSenderImpl internal constructor(
 
                 is Conversation.ProtocolInfo.Proteus -> {
                     // TODO(messaging): make this thread safe (per user)
-                    attemptToSendWithProteus(message)
+                    attemptToSendWithProteus(message, targetRecipients)
                 }
             }
         }
-}
+    }
 
-    private suspend fun attemptToSendWithProteus(message: Message.Regular): Either<CoreFailure, String> {
+    private suspend fun attemptToSendWithProteus(
+        message: Message.Regular,
+        targetRecipients: List<Recipient>?
+    ): Either<CoreFailure, String> {
         val conversationId = message.conversationId
-        return conversationRepository.getConversationRecipients(conversationId).flatMap { recipients ->
+
+        return targetRecipients?.let {
+            messageEnvelopeCreator.createOutgoingEnvelope(it, message).flatMap { envelope ->
+                trySendingProteusEnvelope(envelope, message)
```

Comment:
No need to call `sessionEstablisher.prepareRecipientsForNewOutgoingMessage(recipients)`?

Reply:
We actually do!

Reply:
Can you explain why because I don't see it.